pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -347,14 +321,20 @@ func (plugin *kubenetNetworkPlugin) SetUpPod(namespace string, name string, id k
 	shaper := plugin.shaper()
 
 	if egress != nil || ingress != nil {
-		ipAddr := plugin.podIPs[id]
-		if err := shaper.ReconcileCIDR(fmt.Sprintf("%s/32", ipAddr), egress, ingress); err != nil {
+		if err := shaper.ReconcileCIDR(fmt.Sprintf("%s/32", ip4.String()), egress, ingress); err != nil {
 			return fmt.Errorf("Failed to add pod to shaper: %v", err)
 		}
 	}
 
-	plugin.syncHostportsRules()
-	return nil
+	plugin.podIPs[id] = ip4.String()
+
+	// Open any hostports the pod's containers want
+	runningPods, err := plugin.getRunningPods()
+	if err == nil {
+		err = plugin.hostportHandler.OpenPodHostportsAndSync(pod, BridgeName, runningPods)
```

Comment:
Agreed. That is mostly because network plugin is event triggered. 

I was thinking adding a  Reconcile interface into network plugin. It can be triggered periodically and do whatever it want to ensure/cleanup network configuration.